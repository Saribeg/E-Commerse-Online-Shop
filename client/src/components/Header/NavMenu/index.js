import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import {
  getNavMenuItems,
  openSubMenu,
  closeSubMenu
} from "../../../actions/navMenuActions";

import Preloader from "../../Preloader";

// import "../../../scss/style.scss";
import "./navMenu.scss";

class NavMenu extends Component {
  // Calling action-creator for getting from server object with Navigation Menu Items
  componentDidMount() {
    this.props.getNavMenuItems();
  }

  render() {
    // Creating the category list (men, women)
    let menuList = this.props.navMenuItems.map(e => {
      return (
        <li
          className="main-menu-item"
          key={e._id}
          onMouseOver={() => this.props.openSubMenu(e.categoryName)}
        >
          <Link to={e.categoryUrl} className="main-menu-link">
            {e.categoryName}
          </Link>
        </li>
      );
    });

    // Creating list of subcategories (e.g. clothing) and further subcategories (e.g. shirts, pants)
    let subMenuList = this.props.navMenuItems.map(category => {
      if (category.categoryName === this.props.currentOnMouseOverCategory) {
        return category.subCategoryList.map(subCategory => {
          // further subcategories (e.g. shirts, pants)
          let subfurtherSubCategory = subCategory.furtherSubCategoryList.map(
            furtherSubCategory => {
              return (
                <li
                  className="sub-menu-category-item"
                  key={furtherSubCategory._id}
                >
                  <Link
                    to={furtherSubCategory.furtherSubCategoryUrl}
                    className="sub-menu-category-link"
                  >
                    {furtherSubCategory.furtherSubCategoryName}
                  </Link>
                </li>
              );
            }
          );

          // subcategories (e.g. clothing)
          return (
            <div className="sub-menu-left-list" key={subCategory._id}>
              <Link
                to={subCategory.subCategoryUrl}
                className="sub-menu-left-title"
              >
                {subCategory.subCategoryName.charAt(0).toUpperCase() +
                  subCategory.subCategoryName.slice(1)}
              </Link>
              <ul className="sub-menu-category-list">
                {subfurtherSubCategory}
              </ul>
            </div>
          );
        });
      }
      return null; // to return a value at the end of arrow function as it is expected
    });

    let subMenuSection = (
      <section
        className="sub-menu-wrapper"
        onMouseLeave={this.props.closeSubMenu}
      >
        <div className="container">
          <div className="sub-menu-wrapper-inner">
            <div className="sub-menu-left">{subMenuList}</div>
          </div>
        </div>
      </section>
    );

    // Rendering the whole component
    return (
      <Fragment>
        <ul className="main-menu-list">
          {this.props.isMenuFetching ? <Preloader /> : menuList}
        </ul>

        {this.props.navMenuWindowStatus ? subMenuSection : null}
      </Fragment>
    );
  }
}

const mapStateToProps = state => {
  return {
    navMenuItems: state.navMenu.navMenuItems,
    isMenuFetching: state.navMenu.isMenuFetching,
    navMenuWindowStatus: state.navMenu.navMenuWindowStatus,
    currentOnMouseOverCategory: state.navMenu.currentOnMouseOverCategory
  };
};

export default connect(
  mapStateToProps,
  { getNavMenuItems, openSubMenu, closeSubMenu }
)(NavMenu);
