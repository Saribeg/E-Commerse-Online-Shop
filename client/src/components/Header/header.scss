@import "../../scss/reset.scss";
@import "../../scss/resources/resources.scss";

.main-header {
  border-bottom: 1px solid #ccc;
  position: relative;
}

@media (max-width: 992px) {
  .main-header {
    display: none;
  }
}

.header-content {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 24px 0;
}

@media (max-width: 992px) {
  .header-content {
    padding: 10px 0;
  }
}

.header-burger-menu {
  display: none;
}

@media (max-width: 992px) {
  .header-burger-menu {
    display: block;
    background: url("../../img/burger-menu.png") no-repeat;
    width: 26px;
    height: 21px;
    margin-right: 10px;
  }
}

.logo {
  width: 120px;
  height: 22px;
  margin-right: 41px;
}

@media (max-width: 992px) {
  .logo {
    margin-right: 0;
  }
}
