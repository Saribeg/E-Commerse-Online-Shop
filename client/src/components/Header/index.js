import React, { Component, Fragment } from "react";

import {NavLink} from 'react-router-dom'

import TopBlockAuth from "../TopBlockAuth";
import NavMenu from "./NavMenu";

class Header extends Component {
  render() {
    return (
      <header className="main-header">
        <div className="container">
          <div className="header-content">
            <div className="header-burger-menu" />
            <NavLink to="/">
                <img src="../img/matter.png" alt="Matter" />
            </NavLink>

            <TopBlockAuth />
              <NavMenu />

          </div>
        </div>
      </header>
    );
  }
}

export default Header;
