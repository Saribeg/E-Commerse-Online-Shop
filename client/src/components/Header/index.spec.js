import React from 'react';
import Header from './index';

describe('Header component', () => {
    it('Header component render', () => {
        const wrapper  = shallow(<Header />)
        expect(wrapper).toMatchSnapshot()
    })
})