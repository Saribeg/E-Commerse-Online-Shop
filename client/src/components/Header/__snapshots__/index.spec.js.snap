// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="header-content"
        >
          <div
            className="header-burger-menu"
          />
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="logo"
            to="/"
          >
            <img
              alt="Matter"
              src="matter.svg"
            />
          </NavLink>
          <Connect(NavMenu) />
          <Connect(withRouter(TopBlockAuth)) />
        </div>
      </div>,
      "className": "main-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-content"
        >
          <div
            className="header-burger-menu"
          />
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="logo"
            to="/"
          >
            <img
              alt="Matter"
              src="matter.svg"
            />
          </NavLink>
          <Connect(NavMenu) />
          <Connect(withRouter(TopBlockAuth)) />
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-burger-menu"
            />,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="logo"
              to="/"
            >
              <img
                alt="Matter"
                src="matter.svg"
              />
            </NavLink>,
            <Connect(NavMenu) />,
            <Connect(withRouter(TopBlockAuth)) />,
          ],
          "className": "header-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-burger-menu",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <img
                alt="Matter"
                src="matter.svg"
              />,
              "className": "logo",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Matter",
                "src": "matter.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="header-content"
          >
            <div
              className="header-burger-menu"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="logo"
              to="/"
            >
              <img
                alt="Matter"
                src="matter.svg"
              />
            </NavLink>
            <Connect(NavMenu) />
            <Connect(withRouter(TopBlockAuth)) />
          </div>
        </div>,
        "className": "main-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header-content"
          >
            <div
              className="header-burger-menu"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="logo"
              to="/"
            >
              <img
                alt="Matter"
                src="matter.svg"
              />
            </NavLink>
            <Connect(NavMenu) />
            <Connect(withRouter(TopBlockAuth)) />
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header-burger-menu"
              />,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="logo"
                to="/"
              >
                <img
                  alt="Matter"
                  src="matter.svg"
                />
              </NavLink>,
              <Connect(NavMenu) />,
              <Connect(withRouter(TopBlockAuth)) />,
            ],
            "className": "header-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header-burger-menu",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <img
                  alt="Matter"
                  src="matter.svg"
                />,
                "className": "logo",
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Matter",
                  "src": "matter.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
